tests/integration/test_tickets.py

```diff
@@ -1,134 +1,122 @@
-    assert response.json == {"message": "Tickets successfully deleted"}
-
-    missingIds = {"notIds": [5]}
-    response = client.delete(endpoint, json=missingIds, headers=auth_headers["pm"])
-    assert is_valid(response, 400)
-    assert response.json == {"message": "Ticket IDs missing in request"}
+from models.tickets import TicketModel
+from schemas.ticket import TicketSchema
+from unittest.mock import patch
+
+
+@pytest.mark.usefixtures("client_class", "empty_test_db")
+class BaseConfig:
+    endpoint = "/api/tickets"
+
+
+class TestTicketsGET(BaseConfig):
+    def test_tickets_get_all(self, valid_header, create_ticket):
+        ticket_1 = create_ticket()
+        ticket_2 = create_ticket()
+        response = self.client.get(self.endpoint, headers=valid_header)
+
+        assert response.status_code == 200
+        assert response.json == {"tickets": [ticket_1.json(), ticket_2.json()]}
+
+    def test_tickets_get_by_tenant(self, valid_header, create_ticket):
+        ticket_1 = create_ticket()
+        create_ticket()
+        create_ticket()
+
+        response = self.client.get(
+            f"{self.endpoint}?tenant_id={ticket_1.tenant.id}", headers=valid_header
+        )
+
+        assert response.status_code == 200
+        assert response.json == {"tickets": [ticket_1.json()]}
+
+    def test_tickets_get_one(self, valid_header, create_ticket):
+        ticket = create_ticket()
+        response = self.client.get(f"{self.endpoint}/{ticket.id}", headers=valid_header)
+
+        assert response.status_code == 200
+        assert response.json == ticket.json()
+
+
+class TestTicketsPOST(BaseConfig):
+    def test_create_ticket(self, valid_header):
+        new_ticket = {
+            "author_id": 1,
+            "tenant_id": 1,
+            "status": "New",
+            "urgency": "low",
+            "issue": "Lead paint issue",
+        }
+
+        with patch.object(TicketModel, "create") as mock_create:
+            response = self.client.post(
+                self.endpoint, json=new_ticket, headers=valid_header
+            )
+
+        mock_create.assert_called_once_with(schema=TicketSchema, payload=new_ticket)
+
+        assert response.status_code == 201
+        assert response.json == {"message": "Ticket successfully created"}
+
+
+class TestTicketsPUT(BaseConfig):
+    def test_update_ticket(self, valid_header, create_ticket):
+        ticket = create_ticket()
+        with patch.object(TicketModel, "update", return_value=ticket) as mock_update:
+            response = self.client.put(
+                f"{self.endpoint}/{ticket.id}",
+                json={"hello": "world"},
+                headers=valid_header,
+            )
+
+        mock_update.assert_called_once_with(
+            schema=TicketSchema, id=ticket.id, payload={"hello": "world"}
+        )
+
+        assert response.status_code == 200
+        assert response.json == ticket.json()
+
+
+class TestTicketsDELETE(BaseConfig):
```

Comment:
Same as what I said above for the get endpoint. Let's split these tests to separate the different methods we're testing. One class for each method.